js/modules/k6/experimental/fs/file.go

```diff
@@ -38,14 +39,16 @@ type FileInfo struct {
 //
 // If the end of the file has been reached, it returns EOFError.
 func (f *file) Read(into []byte) (n int, err error) {
-	start := f.offset
-	if start == len(f.data) {
+	offset := f.offset.Load()
+
+	start := offset
+	if start == f.size() {
 		return 0, newFsError(EOFError, "EOF")
 	}
 
-	end := f.offset + len(into)
-	if end > len(f.data) {
-		end = len(f.data)
+	end := offset + int64(len(into))
+	if end > f.size() {
+		end = f.size()
```

Comment:
```suggestion
	if end < start + len(into) {
		end = start + int64(len(into))
```

Reply:
I have revamped the function completely to make it more readable and cater to your comment 🙇🏻